src/common/notifications/notifications.cpp

```diff
@@ -206,6 +206,9 @@ void notifications::show_toast(std::wstring message, std::wstring title, toast_p
     show_toast_with_activations(std::move(message), std::move(title), {}, {}, std::move(params));
 }
 
+// Disable 26497 this function should be evaluated at compile time
```

Comment:
Why?

Reply:
The function is void and has a side effect.
I can't see how to do data.swap(buffer) at compile time.

Reply:
The warning disable makes sense, but on each function that needs this disabled, you should add comments saying why it needs to be disabled for that function. Please change the PR accordingly.

Reply:
While trying to write a good reason why it is marked as a constexpr candidate by the compiler, I just find out that C++20 has a constexpr version of the "swap". So I will mark the function constexpr, instead of disabling the warning.
the constexpr version is used if possible, if not, a compile-time one is used.

Reply:
Does it really work as constexpr at runtime?
The comment applies to the other changes in this PR as well (excluding external headers, of course):
> on each function that needs this disabled, you should add comments saying why it needs to be disabled for that function

Reply:
constexpr only work as constexpr at compile time, if it is not possible it uses the non-constexpr version of the function.
So I am making all constexpr, even if in the code we only use it at run time.